File: CYJmission01.py
import RPi.GPIO as GPIO
import time

# GPIO 핀 번호 설정
LED_RED = 7
LED_GREEN = 21
LED_BLUE = 22

# GPIO 초기화
GPIO.setmode(GPIO.BCM)
GPIO.setup(LED_RED, GPIO.OUT)
GPIO.setup(LED_GREEN, GPIO.OUT)
GPIO.setup(LED_BLUE, GPIO.OUT)

try:
    print("3 Color LED Control Start !!")
    for i in range(10):
        GPIO.output(LED_RED, GPIO.HIGH)
        GPIO.output(LED_GREEN, GPIO.HIGH)
        GPIO.output(LED_BLUE, GPIO.HIGH)
        time.sleep(0.5)
        
        GPIO.output(LED_RED, GPIO.LOW)
        GPIO.output(LED_GREEN, GPIO.LOW)
        GPIO.output(LED_BLUE, GPIO.LOW)
        time.sleep(0.5)

except KeyboardInterrupt:
    pass

# GPIO 리소스 정리
GPIO.cleanup()

File: CYJmission01.c
#include <stdio.h>
#include <wiringPi.h>
#include <unistd.h>

#define LED_RED 7
#define LED_GREEN 21
#define LED_BLUE 22

int main(void) {
    if (wiringPiSetup() == -1)
        return 1;

    pinMode(LED_RED, OUTPUT);
    pinMode(LED_GREEN, OUTPUT);
    pinMode(LED_BLUE, OUTPUT);

    digitalWrite(LED_RED, 0);
    digitalWrite(LED_GREEN, 0);
    digitalWrite(LED_BLUE, 0);

    printf("3 Color LED Control Start !!\n");

    for (int i = 0; i < 10; i++) {
        digitalWrite(LED_RED, 1);
        digitalWrite(LED_GREEN, 1);
        digitalWrite(LED_BLUE, 1);
        usleep(500000);

        digitalWrite(LED_RED, 0);
        digitalWrite(LED_GREEN, 0);
        digitalWrite(LED_BLUE, 0);
        usleep(500000);
    }

    return 0;
}

